native/membrane_videocompositor/src/elixir_bridge/elixir_structs.rs

```diff
@@ -136,3 +140,710 @@ impl From<ElixirCropping> for Box<dyn TextureTransformation> {
         ))
     }
 }
+
+#[derive(Debug, Clone, Copy, rustler::NifStruct)]
+#[module = "Membrane.VideoCompositor.Scene.Resolution"]
+pub struct Resolution {
+    width: u32,
+    height: u32,
+}
+
+#[derive(Debug, rustler::NifStruct)]
+#[module = "Membrane.VideoCompositor.Scene.Object.Texture.RustlerFriendly"]
+pub struct Texture {
+    input: ObjectName,
+    // this is a placeholder only temporarily, until we figure out encoding transformations
+    // then, it will be something like `Arc<dyn Transformation>`
+    transformations: Vec<ImplementationPlaceholder>,
+    resolution: TextureOutputResolution,
+}
+
+impl Texture {
+    fn mentioned_names(&self) -> Vec<&Name> {
+        let mut names = vec![&self.input];
+
+        if let TextureOutputResolution::Name(name) = &self.resolution {
+            names.push(name);
+        }
+
+        names
+    }
+}
```

Comment:
Since we implement the same `mentioned_names` function in `Texture` and `Layout`, I think that we should define `Trait` for that and implement it here

Reply:
it doesn't really make sense, since it wouldn't simplify any code. This would be a good solution if we wanted `Object` to not be an `enum` and hold a Box<dyn Sth> instead. Using a trait doesn't fit here.